### PR TITLE
Serdes v2: Make v2 the default. Pass `--legacy true` to do v1.

### LINKED ISSUE #25935: Rename Serialization commands and change the default version
## Context
- As we ship v2, we are keeping v1 available to facilitate migrations.
- The `dump` and `load` command names are ambiguous, and we can improve them.
- We need to communicate that v1 will be deprecated in 46.

### Import and Export
- Create two new commands `export` and `import`.
- `export` replaces `dump` and `import` replaces `load`.
- `export` and `import` default to Serdes v2. It is not possible to use legacy with these commands.

### Dump and Load
- We keep `dump` and `load` and they default to Serdes v1.
- Users can use v2 by adding the `--v2 true` flag.
- We add a red and bold warning that the `dump` and `load` commands will be deprecated and removed in 46.

> The `dump` and `load` commands are deprecated and will be removed in the 46 release. Please start using `export` and `import`. Note that the new commands have different behavior. Read the docs for a description of all changes.

### PR DESCRIPTION
**Pull request by bshepherdson**
Fixes #25935

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/26206)

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj [hunk 1]
```diff
@@ -80,10 +87,14 @@
 
 (defn load
   "Load serialized metabase instance as created by `dump` command from directory `path`."
-  [path args]
-  (if (:v2 args)
-    (v2-load path args)
-    (v1-load path args)))
+  [path {:keys [legacy v2] :as args}]
+  (prn "load" legacy v2 args)
```
Comment (qnkhuat) [inline enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:91]: probably should remove this?

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj [hunk 2]
```diff
@@ -80,10 +87,14 @@
...
+  (when (and v2 legacy)
+    (log/error (trs "ERROR load cannot be both --legacy and --v2"))
+    (throw (ex-info "load cannot be both --legacy and --v2" {})))
```
Comment (qnkhuat) [inline enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:94]: should we `i18n` the exception too?

**Review by qnkhuat — APPROVED** [review]
LGTM, just minor comments.

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj (hunk 1 above)
Comment (bshepherdson) [reply enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:91]: Done.

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj (hunk 2 above)
Comment (bshepherdson) [reply enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:94]: Done.

**Review by metamben — APPROVED** [review]

**Review by camsaul — APPROVED** [review]

**Comment by bshepherdson** [comment]
I'm deliberately sitting on this until Serdes v2 is where we want it to be for 45.
We don't want to point the default at something that isn't working fully.

**Pull request closed without merging**